### PR TITLE
Audit two-dimensional-arrays.md

### PR DESCRIPTION
**Pull request by DerekJxy**
Refer to the issue #88 
What I've done:
+ Break sentences with one space instead of two spaces
+ Add alt text to the image
+ Reset the indentation of all code examples
+ Remove all <br/ >
+ Improve the readability(Highlight some important reading words)
+ Adjust the for loop code

Things I can't accomplish:
+ Can't set the image to looking good in dark mode
+ Can't set the commented lines in code example perform good in dark mode

### CODE REVIEW

**Review by sirinoks — APPROVED** [review]
Solid formatting improvements

**Comment by sirinoks** [comment]
I have went ahead and fixed an image used to one that will work with dark mode. This is an example of how the new one will look like on a dark background:


And here is the final image that has transparency and will work with dark or while background:


You might want to choose a different host for these images, since those links are hosted through Discord right now.
_Black letters with a white stroke will always work_

File: docs/F-Refinements/two-dimensional-arrays.md [hunk 1]
```diff
@@ -7,22 +7,22 @@ sidebar_position: 5
...
-This chapter introduces two-dimensional arrays, describing their syntax and their organization in memory.  This chapter includes sample code snippets related to arrays of character strings. 
+This chapter introduces two-dimensional arrays, describing their `syntax` and their `organization` in memory. This chapter includes sample code snippets related to arrays of character strings. 
 
 ## Two-Dimensional Syntax
 
-A table is a useful analogy for describing a two-dimensional array.  An entry in a table is identified by its row and column positions.  Consider the row and column indices in the figure below.  The first index refers to the row and the second index refers to the column.
+A table is a useful analogy for describing a two-dimensional array. An entry in a table is identified by its row and column positions. Consider the row and column indices in the figure below. `The first index` refers to the `row` and `the second index` refers to the `column`.
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/2d%20array.png)
+![2dArrayExample](https://ict.senecacollege.ca//~ipc144/pages/images/2d%20array.png)
```
Comment (Reza9472) [inline docs/F-Refinements/two-dimensional-arrays.md:23]: In discussion with project maintainers, I think they agree on having more descriptive alt tags for the images.
Ex: [a 5x5 table, describing two-dimensional arrays

**Review by Reza9472 — CHANGES_REQUESTED** [review]
Great Work on these changes!
Please check this small change as well

File: docs/F-Refinements/two-dimensional-arrays.md [hunk 2]
```diff
@@ -7,22 +7,22 @@ sidebar_position: 5
```
Comment (Reza9472) [inline docs/F-Refinements/two-dimensional-arrays.md:7]: fix Frontmatter for page to include proper id, title, slug, etc
Possible solution can be

---
id: two-dimensional-arrays
title: Two-Dimensional Arrays
sidebar_position: 5
slug: /refinements/two-dimensional-arrays
description: This chapter describes the use of two-dimensional arrays 
---

**Review by Reza9472 — CHANGES_REQUESTED** [review]
One thing that you have forgotten here is fixing the frontmatter for this page.
Please check the following review:

File: docs/F-Refinements/two-dimensional-arrays.md [hunk 3]
```diff
@@ -84,14 +84,14 @@ a[2][0] a[2][1] a[2][2] a[2][3] a[2][4]
 a[3][0] a[3][1] a[3][2] a[3][3] a[3][4] 
 ```
 
-> **Warning**<br/>
+> **Warning**
```
Comment (abatomunkuev) [inline docs/F-Refinements/two-dimensional-arrays.md:87]: Suggestion to add an admonition. For example, you may use the following admonition -> `:::caution`

File: docs/F-Refinements/two-dimensional-arrays.md [hunk 4]
```diff
@@ -267,22 +269,23 @@ The above example references the address of the third character within the secon
...
 
-for (i = 0; i < 5; i++)
-        scanf(" %[^\n]", name[i]); 
+for (i = 0; i < 5; i++){
+    scanf(" %[^\n]", name[i]); 
+}
 ```
 
-> **Reminder**<br/>
+> **Reminder**
```
Comment (abatomunkuev) [inline docs/F-Refinements/two-dimensional-arrays.md:285]: Suggestion to add an admonition. For example, you may use the following -> `:::note` or `:::info`

**Review by abatomunkuev — COMMENTED** [review]
Hello! I have reviewed the changes. Please consider my comments to improve the markdown file.

**Comment by DerekJxy** [comment]
> Solid formatting improvements



> I have went ahead and fixed an image used to one that will work with dark mode. This is an example of how the new one will look like on a dark background: 
> 
> And here is the final image that has transparency and will work with dark or while background: 
> 
> You might want to choose a different host for these images, since those links are hosted through Discord right now. _Black letters with a white stroke will always work_

Hi, can you make it more specific? I tried to add it directly, but that doesn't seems like it supposed to be. How can I make these pictures work in the md file?

File: docs/F-Refinements/two-dimensional-arrays.md (hunk 2 above)
Comment (DerekJxy) [reply docs/F-Refinements/two-dimensional-arrays.md:7]: Nice work! That's a good suggestion!

File: docs/F-Refinements/two-dimensional-arrays.md (hunk 4 above)
Comment (DerekJxy) [reply docs/F-Refinements/two-dimensional-arrays.md:285]: Great! That will make the md file looks nicer!

**Comment by sirinoks** [comment]
> Hi, can you make it more specific? I tried to add it directly, but that doesn't seems like it supposed to be. How can I make these pictures work in the md file?

You literally put an `` html element with a link to the image between the double quotes. OR text explanation

Perhaps the image URLs aren't working for you. Here is the URL of the one with transparency directly: https://cdn.discordapp.com/attachments/353432193730871296/911171345587503114/68747470733a2f2f6963742e73656e656361636f6c6c6567652e63612f2f7e6970633134342f70616765732f696d616765732f326425323061727261792e706e67.png

**Comment by DerekJxy** [comment]
> > Hi, can you make it more specific? I tried to add it directly, but that doesn't seems like it supposed to be. How can I make these pictures work in the md file?
> 
> You literally put an `` html element with a link to the image between the double quotes. OR `text explanation`
> 
> Perhaps the image URLs aren't working for you. Here is the URL of the one with transparency directly: https://cdn.discordapp.com/attachments/353432193730871296/911171345587503114/68747470733a2f2f6963742e73656e656361636f6c6c6567652e63612f2f7e6970633134342f70616765732f696d616765732f326425323061727261792e706e67.png



> > Hi, can you make it more specific? I tried to add it directly, but that doesn't seems like it supposed to be. How can I make these pictures work in the md file?
> 
> You literally put an `` html element with a link to the image between the double quotes. OR `text explanation`
> 
> Perhaps the image URLs aren't working for you. Here is the URL of the one with transparency directly: https://cdn.discordapp.com/attachments/353432193730871296/911171345587503114/68747470733a2f2f6963742e73656e656361636f6c6c6567652e63612f2f7e6970633134342f70616765732f696d616765732f326425323061727261792e706e67.png

I tried to use this `text explanation`  with the link you provided to the md file, but the result was the same as it used to be.(Hard to see the word `rows` and `columns` in dark mode) Probably your first way would work.

**Comment by sirinoks** [comment]
Oh sorry! I gave you a wrong link. My bad. Here is the correct one - https://cdn.discordapp.com/attachments/353432193730871296/911315058041171988/table_any.png

**Comment by DerekJxy** [comment]
> Oh sorry! I gave you a wrong link. My bad. Here is the correct one - https://cdn.discordapp.com/attachments/353432193730871296/911315058041171988/table_any.png

Okay! Great! This one works perfectly!

**Review by CameronGray1210 — APPROVED** [review]
Good thanks!

**Comment by DerekJxy** [comment]
> Good thanks!

Thank you!

**Pull request merged**